geode-core/src/main/java/org/apache/geode/management/internal/cli/CliUtil.java

```diff
@@ -183,6 +185,31 @@ public static String getMemberNameOrId(DistributedMember distributedMember) {
         cache.getDistributionManager().getNormalDistributionManagerIds());
   }
 
+  /**
+   * Returns a set of all the members of the distributed system of a specific version excluding
+   * locators.
+   */
+  @SuppressWarnings("unchecked")
+  public static Set<DistributedMember> getNormalMembersWithSameOrNewerVersion(InternalCache cache,
+      Version version) {
+    return getAllNormalMembers(cache).stream().filter(
+        member -> ((InternalDistributedMember) member).getVersionObject().compareTo(version) >= 0)
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Returns a set of all the members of the distributed system older than a specific version
+   * excluding locators.
+   */
+  @SuppressWarnings("unchecked")
+  public static Set<DistributedMember> getNormalMembersWithSameOrOlderVersion(InternalCache cache,
+      Version version) {
+    return getAllNormalMembers(cache).stream()
+        .filter(member -> ((InternalDistributedMember) member).getVersionObject().equals(version))
```

Comment:
The method is named SameOrOlderVersion but this filter is only filtering for equals?

Reply:
This may return new members